File: subprojects/docs/src/snippets/plugins/namedDomainObjectContainer/kotlin/buildSrc/src/main/java/DownloadExtension.java
import org.gradle.api.NamedDomainObjectContainer;
import org.gradle.api.model.ObjectFactory;

import javax.inject.Inject;

public class DownloadExtension {
    // A container of `Resource` objects
    private final NamedDomainObjectContainer<Resource> resources;

    @Inject
    public DownloadExtension(ObjectFactory objectFactory) {
        // Use an injected ObjectFactory to create a container
        resources = objectFactory.domainObjectContainer(Resource.class);
    }

    public NamedDomainObjectContainer<Resource> getResources() {
        return resources;
    }
}

File: subprojects/docs/src/snippets/plugins/namedDomainObjectContainer/kotlin/buildSrc/src/main/kotlin/DownloadExtension.kt

import javax.inject.Inject
import org.gradle.api.model.ObjectFactory
import org.gradle.api.NamedDomainObjectContainer

open class DownloadExtension @Inject constructor(objectFactory: ObjectFactory) {
    // Use an injected ObjectFactory to create a container of `Resource` objects
    val resources = objectFactory.domainObjectContainer(Resource::class.java)
}
